main.tf

```diff
@@ -1,105 +1,90 @@
-# Create a VPC
-resource "aws_vpc" "main" {
-  cidr_block = "10.0.0.0/16"
+#declaring locals
+
+locals {
+
+  vpc_cidr           = "10.0.0.0/16"
+  public_cidr        = ["10.0.0.0/24", "10.0.1.0/24"]
+  private_cidr       = ["10.0.2.0/24", "10.0.3.0/24"]
+  availability_zones = ["us-east-1a", "us-east-1b"]
 
-  tags = {
-      Name = "terraform-launched-vpc"
-  }
 
 }
 
-#creating public subnet 1
-resource "aws_subnet" "public-subnet-1" {
-  vpc_id     = aws_vpc.main.id
-  cidr_block = "10.0.0.0/24"
+
+# Create a VPC
+resource "aws_vpc" "main" {
+  cidr_block = local.vpc_cidr
 
   tags = {
-      Name = "public subnet 1"
+    Name = "terraform-launched-vpc"
   }
 
 }
 
+#creating public subnets
+resource "aws_subnet" "public" {
+
+  count = 2
```

Comment:
could you replace the number 2 with length(local.public_cidr) for resources related to the public subnet and use
length(local.private_cidr) for resources related to the private subnet as it makes the code more neat.
i.e `count = length(local.public_cidr)` or `count = length(local.private_cidr)`